src/backend/expungeservice/endpoints/auth.py

```diff
@@ -41,8 +41,10 @@ def wrapper(*args, **kwargs):
                 current_app.config.get('JWT_SECRET_KEY')
             )
 
-            # TODO temp hack - replace with table from DB
-            if not payload['sub'] in users:
+            g.auth_hdr_payload = payload
+
+            user_db_result = get_user_by_email(g.database, payload['sub'])
```

Comment:
Maybe this is out of scope for this PR, but it's best practice to use a immutable, numeric ID as the primary key for users at all times.

(The only exception to this is when someone logs in and you have to look them up with their email address.)

Given the nature of this project, maybe this is fine (maybe we don't allow users to change their emails) but at least this should be commented somewhere (if it's not already) as a form of technical debt.

Reply:
@maxkwallace I think that's correct. The initial lookup upon login has to be by email or username (that's the only info we have about the user) but after the password check the user_id should go in the token and on subsequent requests authentication should be done by decoding the token and checking user_id.

Reply:
Another point to make here is that `jwt.decode` can throw a number of exceptions and those exceptions should be checked when decoding the token. See this page for more info: https://pyjwt.readthedocs.io/en/latest/api.html

Reply:
Ah, after looking at the full file I see you are checking for exceptions. Never mind!

Reply:
I think my refactor addresses this: it no longer uses email as an identifier except in the auth_token GET function (I realized this function is missing the check for required data fields though. Will add that.)